### PR TITLE
Update NLPModelsJuMP for JuMP 0.22

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -11,8 +11,8 @@ Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 
 [compat]
-JuMP = "0.21"
-MathOptInterface = "0.9, 0.10"
+JuMP = "0.22"
```
Comment (dpo) [inline Project.toml:14]: Can we keep using 0.21 or not?

Comment (amontoison) [reply Project.toml:14]: Unfortunately no, JuMP 0.22 is a breaking release.

**Review by tmigot — APPROVED** [review]

**Comment by tmigot** [comment]
Following [PR78 in `OptimizationProblems.jl`](https://github.com/JuliaSmoothOptimizers/OptimizationProblems.jl/pull/78), you will have to update the docs environment after `OptimizationProblems.jl` gets a new release.

**Comment by amontoison** [comment]
Let's merge this PR and do a new release.
The documentation build cannot work before the support of JuMP 0.22 with OptimizationProblems.jl.

**Pull request merged**